java-checks/src/main/java/org/sonar/java/checks/serialization/SerializableFieldInSerializableClassCheck.java

```diff
@@ -54,14 +60,46 @@ public void visitNode(Tree tree) {
     if (isSerializable(classTree)
       && !SerializableContract.hasSpecialHandlingSerializationMethods(classTree)
       && !classTree.symbol().type().isSubtypeOf("javax.servlet.http.HttpServlet")) {
-      for (Tree member : classTree.members()) {
-        if (member.is(Tree.Kind.VARIABLE)) {
-          checkVariableMember((VariableTree) member);
-        }
-      }
+      
+      Set<String> constructorInjectedParams = getConstructorInjectedFields(classTree);
+      Set<String> setterInjectedParams = getSetterInjectedFields(classTree);
+      
+      classTree.members().stream()
+        .filter(member -> member.is(Tree.Kind.VARIABLE))
+        .map(VariableTree.class::cast)
+        .filter(variableTree -> 
```

Comment:
We are doing a first filtering here, and then a second in `checkVariableMember` thanks to `isExcluded`. We could move the second filtering here.
It will avoid having the differents kind of injections being filtered at two different places.